packages/database/src/migrations/20210607020237-AddIndicatorsForAuFlutracking-modifies-data.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+import { deleteObject, generateId, insertObject } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const INDICATOR = {
+  id: generateId(),
+  code: 'FluTracker_LGA_Percent_First_Nations_ILI',
+  builder: 'analyticArithmetic',
+  config: {
+    formula:
+      'firstExistingValue(FWV_LGA_ILI_Percentage_prior_March_2021, FWV_LGA_ILI_Percentage_Till_Current)',
+    parameters: [
```

Comment:
While nested indicator parameters is a really cool design tool, and allows for us to create easily readable indicator formulas as you've done here, right now we haven't really optimised the performance of them so they should be used sparingly. I think given that each parameter here is using a unique dataElementCode means we can get away with just a shallow indicator, such as:
```
  config: {
    formula:
      'firstExistingValue(FluTracking_NationalFNILI, FWV_LGA_003a=== 0 ? 0 : FWV_LGA_004a / FWV_LGA_003a)',
    aggregation: {
      FluTracking_NationalFNILI: 'FINAL_EACH_DAY',
      FWV_LGA_003a: ['FINAL_EACH_DAY', {
        type: 'SUM_PER_PERIOD_PER_ORG_GROUP',
        config: {
          dataSourceEntityType: 'sub_district',
          aggregationEntityType: 'country',
        },
      } ],
      FWV_LGA_004a: ['FINAL_EACH_DAY', {
        type: 'SUM_PER_PERIOD_PER_ORG_GROUP',
        config: {
          dataSourceEntityType: 'sub_district',
          aggregationEntityType: 'country',
        },
      } ],
    },
    defaultValues: {
      FluTracking_NationalFNILI: 'undefined',
      FWV_LGA_003a: 'undefined',
      FWV_LGA_004a: 'undefined'
    }
  }
```

Reply:
I have tried but seems like `firstExistingValue()` doesn't support expression in the parameters. I agree it would be nice to have this function.

We can keep moving things forward to use parameters. For that new function do you want me to make a card? @rohan-bes

Reply:
^ignore me